tests/json_test.cpp

```diff
@@ -170,5 +170,38 @@ void JsonUnsortedArrayTest() {
   TEST_NOTNULL(monster->testarrayoftables()->LookupByKey("ccc"));
 }
 
+void JsonUnionStructTest() {
+  // schema to parse data
+  auto schema = R"(
+struct MyStruct { field: int; }
```

Comment:
odd identing, could you fix it a bit?

Reply:
You can do that in a follow up PR.

Reply:
@dbaileychess 
The indenting seems to be odd because it is a multiline string literal. Same for json_source from line 182. If you still want to change the indentation I propose this:
```cpp
  const char* schema = "struct MyStruct { field: int; }\n"
                       "union UnionWithStruct { MyStruct }\n"
                       "table JsonUnionStructTest { union_with_struct: UnionWithStruct; }\n"
                       "root_type JsonUnionStructTest;\n";
  //...
  const char* json_source = "{\n"
                            "  union_with_struct_type: "MyStruct",\n"
                            "  union_with_struct: {\n"
                            "    field: 12345\n"
                            "  }\n"
                            "}\n";
```

Let me know what you prefer and I will change it as you wish 😊